internal/command/test.go

```diff
@@ -1266,3 +1271,22 @@ func (runner *TestFileRunner) prepareInputVariablesForAssertions(config *configs
 		config.Module.Variables = currentVars
 	}, diags
 }
+
+// checkProblematicPlanErrors checks for plan errors that are also "expected" by the tests. In some cases we expect an error, however,
+// what causes the error might not be what we expected. So we try to warn about that here.
+func checkProblematicPlanErrors(expectedFailures addrs.Map[addrs.Referenceable, bool], planDiags tfdiags.Diagnostics) tfdiags.Diagnostics {
+	for _, diag := range planDiags {
+		if rule, ok := addrs.DiagnosticOriginatesFromCheckRule(diag); ok {
+			if rule.Container.CheckableKind() == addrs.CheckableInputVariable {
+				addr, ok := rule.Container.(addrs.AbsInputVariableInstance)
+				if ok && expectedFailures.Has(addr.Variable) {
+					planDiags = planDiags.Append(tfdiags.Sourceless(
+						tfdiags.Warning,
+						"Invalid Variable in test file",
+						fmt.Sprintf("Variable %s, has an invalid value within the test. Although this was an expected failure, it has meant the apply stage was unable to run so the overall test will fail.", rule.Container.String())))
+				}
+			}
+		}
```

Comment:
nit: would be much more readable if we flatten it a bit:

```suggestion
		rule, ok := addrs.DiagnosticOriginatesFromCheckRule(diag)
		if !ok {
			continue
		}
		if rule.Container.CheckableKind() != addrs.CheckableInputVariable {
			continue
		}
		
		addr, ok := rule.Container.(addrs.AbsInputVariableInstance)
		if ok && expectedFailures.Has(addr.Variable) {
			planDiags = planDiags.Append(tfdiags.Sourceless(
				tfdiags.Warning,
				"Invalid Variable in test file",
				fmt.Sprintf("Variable %s, has an invalid value within the test. Although this was an expected failure, it has meant the apply stage was unable to run so the overall test will fail.", rule.Container.String())))
		}
```